libraries/chain/kv_context_rocksdb.cpp

```diff
@@ -162,9 +165,10 @@ namespace eosio { namespace chain {
             // Not end or at an erased kv pair
             auto kv = kv_it.get_kv();
 
+            EOS_ASSERT(kv->value.size() >= kv_payer_size, kv_rocksdb_bad_value_size_exception , "The size of value returned from RocksDB is less than payer's size");
             // kv is always non-null due to the check of is_valid()
             *found_key_size = kv->key.size();
-            *found_value_size = kv->value.size();
+            *found_value_size = kv->value.size() - kv_payer_size;
```

Comment:
make sure that new static method, to determine actual value size, is called before *found_key_size is set to make sure it doesn't get set without found_value_size if kv_rocksdb_bad_value_size_exception is thrown.

Reply:
Thanks for the reminder.